pandas/core/groupby/ops.py

```diff
@@ -1243,13 +1245,7 @@ def _chop(self, sdata: Series, slice_obj: slice) -> Series:
         # fastpath equivalent to `sdata.iloc[slice_obj]`
         mgr = sdata._mgr.get_slice(slice_obj)
         # __finalize__ not called here, must be applied by caller if applicable
```

Comment:
are we now doing `__finalize__` in all of the places this is called?  if so, better to do it here? (and on L1260)?

Reply:
Yeah, there is a comment there about this:

https://github.com/pandas-dev/pandas/blob/83ea173d150e715790ce7a1b2d329b3eeb4c3ea0/pandas/core/groupby/ops.py#L1263

(from https://github.com/pandas-dev/pandas/pull/37461) 
But I don't know the reason that it was originally decided that finalize must be called by the caller. In any case I don't see a reason to not move it to `_chop` now.

Reply:
> But I don't know the reason that it was originally decided that finalize must be called by the caller.  In any case I don't see a reason to not move it to _chop now.

#37461 OP says it is for perf to not call it on each iteration.  This PR removes that motivation, so go for it!